Zend/zend_inheritance.c

```diff
@@ -1053,9 +1055,19 @@ static void ZEND_COLD emit_incompatible_method_error(
 			}
 		}
 	} else {
-		zend_error_at(E_COMPILE_ERROR, func_filename(child), func_lineno(child),
-			"Declaration of %s must be compatible with %s",
-			ZSTR_VAL(child_prototype), ZSTR_VAL(parent_prototype));
+		if (child->op_array.scope->num_traits) {
+			zend_error_at(E_COMPILE_ERROR, func_filename(child), func_lineno(child),
+				"Declaration of %s (from %s) must be compatible with %s",
+				ZSTR_VAL(child_prototype), ZSTR_VAL(child->op_array.scope->trait_names[0].name), ZSTR_VAL(parent_prototype));
+		} else if (ce && ce != child_scope && ce != parent_scope) {
+			zend_error_at(E_COMPILE_ERROR, func_filename(child), func_lineno(child),
+				"While materializing %s: Declaration of %s must be compatible with %s",
+				ZSTR_VAL(ce->name), ZSTR_VAL(child_prototype), ZSTR_VAL(parent_prototype));
```

Comment:
This branch never seems to trigger in any test.

Reply:
I forgot to commit my test case that shows it. I've now added them.